### DISCUSSION TITLE
Possibility to add local control buttons

### DISCUSSION THREAD

**Discussion by THorst92** [nliaudat/esp32_8ch_motor_shield#67] [Ideas]
Hello,

thanks for this amazing project and all the work!
I thought about to add the possibility to add buttons for a local control. E.g. of the host is down.
I played around with a MCP23017, which can be used for inputs and outputs.
Or maybe add a connector for i2c to add extension like the MCP or a display.

Thorsten

**Comment by nliaudat** [comment]
Hi, 
I was wondering if MCP23017 could be used to extend ADC ? Does it works ?
The main problem of MCP23017 is the price charged at JLCPCB (approx 10$), which is fast the cost of the complete board. 

I will add a i2c connector with 2.54mm pitch to be compatible with dupont cables. 
Regards

**Reply by THorst92** [reply]
Hi,
for binary in and outputs it is working perfectly.
I rebuild a "development" board on base of this project but with the MCP23017 and it worked perfectly. 
From cost side of view I totally understand that you will not use it (10$ are way to expensive).
I found this one for 1.5$ on JLCPCB: [MCP23017-E/SO](https://jlcpcb.com/partdetail/MicrochipTech-MCP23017_ESO/C47023)
Would this work?

Thanks for adding the connector!

Regards.

**Reply by nliaudat** [reply]
![image](https://github.com/user-attachments/assets/1f2e82f1-add2-4598-bc52-6a4cfbb61ac9)
The SMT soldered price is 10.8$.

**Reply by THorst92** [reply]
okay that is too much

**Reply by nliaudat** [reply]
It's better to use a manufactured module like https://fr.aliexpress.com/item/1005005335661057.html

**Reply by mkaiser** [reply]
What about the MCP in the E/SS package?

https://jlcpcb.com/parts/componentSearch?searchTxt=MCP23017-E%2FSS

**Reply by mkaiser** [reply]
> ![image](https://github.com/user-attachments/assets/1f2e82f1-add2-4598-bc52-6a4cfbb61ac9)
> The SMT soldered price is 10.8$. 

Why does the price differ from the jlcpcb Website and from the easyeda Tool  differ that much?
